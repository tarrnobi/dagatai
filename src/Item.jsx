import React, { Component } from 'react';
import PropTypes from 'prop-types';

import { style } from './style';

class Item extends Component {
  constructor(props) {
    super(props);
    // function binds
    this.handleClick = this.handleClick.bind(this);
  }
  handleClick() {
    this.props.itemClicked(this.props.itemDate);
  }
  render() {
    return (
      <li style={style.calendarItem.componentStyle}>
        <button
          style={style.calendarItem.buttonStyle}
          onClick={this.handleClick}
        >
          {this.props.itemDate.getDate()}
        </button>
      </li>);
  }
}

Item.propTypes = {
  itemDate: PropTypes.instanceOf(Date).isRequired,
  itemClicked: PropTypes.func.isRequired,
};

export default Item;
