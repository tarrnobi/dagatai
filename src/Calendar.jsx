import React, { Component } from 'react';
import Item from './Item';
import EntryList from './EntryList';

import { style } from './style';
// const colours = require('./style');
// const style = require('./style').style;

class Calendar extends Component {
  constructor(props) {
    super(props);
    this.state = {
      activeDate: new Date(),
      locale: 'en-US',
      showEntryList: false,
    };
    this.handleChildDaySelection = this.handleChildDaySelection.bind(this);
    this.hideEntryList = this.hideEntryList.bind(this);
  }
  getDaysInMonth() {
    const d = new Date(
      this.state.activeDate.getFullYear(),
      this.state.activeDate.getMonth() + 1, // Jump Ahead One month
      0,
    ); // Last Day of previous Month
    return d.getDate();
  }

  generateCalendarItems() {
    const numDays = this.getDaysInMonth();
    const items = [...Array(numDays)].map((e, i) => {
      const itemDate = new Date(
        this.state.activeDate.getFullYear(),
        this.state.activeDate.getMonth(),
        i + 1,
      );
      return (
        <Item
          key={itemDate}
          itemDate={itemDate}
          itemClicked={this.handleChildDaySelection}
        />
      );
    });
    return items;
  }
  handleChildDaySelection(date) {
    this.setState({
      selectedDate: date,
      showEntryList: true,
    });
  }
  hideEntryList() {
    this.setState({ showEntryList: false });
  }

  render() {
    return (
      <div style={style.container}>
        <header style={style.header}>
          Dagata.io
        </header>
        <div style={style.calendarSection}>
          <h2>{ this.state.activeDate.toLocaleString(this.state.locale, { month: 'long' }) }</h2>
          <ul style={style.datePanel}>
            {this.generateCalendarItems()}
          </ul>
          {this.state.showEntryList === true &&
            <EntryList
              selectedDate={this.state.selectedDate}
              hideEntryList={this.hideEntryList}
              endPoint={`${process.env.REACT_APP_API_HOST}/api/events`}
            />
          }
        </div>
      </div>
    );
  }
}

export default Calendar;
