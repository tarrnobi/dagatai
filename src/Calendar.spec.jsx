import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';

import Calendar from './Calendar';
import Item from './Item';

const MockDate = require('mockdate');

test('Calendar Renders', () => {
  const component = renderer.create(<Calendar />);
  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});

test('Calendar Renders 30 Days for Jun-2017', () => {
  const mockDate = new Date(2017, 5, 1); // Month starts at zero
  MockDate.set(mockDate.valueOf());

  const wrapper = mount(<Calendar />);
  expect(wrapper.find(Item)).toHaveLength(30);
  MockDate.reset();
});

test('Calendar Renders 29 Days for Feb-2016', () => {
  const mockDate = new Date(2016, 1, 1); // Month starts at zero
  MockDate.set(mockDate.valueOf());

  const wrapper = mount(<Calendar />);
  expect(wrapper.find(Item)).toHaveLength(29);
  MockDate.reset();
});
