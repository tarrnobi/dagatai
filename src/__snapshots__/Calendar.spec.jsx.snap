// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calendar Renders 1`] = `
<div
  style={Object {}}
>
  <header
    style={
      Object {
        "backgroundColor": "#AF3A0B",
        "color": "#E37448",
        "fontFamily": "Impact, sans-serif",
        "fontSize": "2em",
        "minHeight": "30px",
      }
    }
  >
    Dagata.io
  </header>
  <div
    style={
      Object {
        "backgroundColor": "#171d22",
        "border": "solid #6B1F00 3px",
      }
    }
  >
    <h2>
      January
    </h2>
    <ul
      style={
        Object {
          "backgroundColor": "#2c3740",
          "display": "inline-block",
          "margin": "0",
          "maxWidth": "50%",
          "padding": "0",
          "verticalAlign": "top",
        }
      }
    >
      <li
        style={
          Object {
            "display": "inline-block",
            "listStyleType": "none",
            "marginBottom": "5px",
            "textAlign": "center",
            "width": "13.6%",
          }
        }
      >
        <button
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "transparent",
              "border": "none",
              "color": "#ffffff",
            }
          }
        >
          1
        </button>
      </li>
      <li
        style={
          Object {
            "display": "inline-block",
            "listStyleType": "none",
            "marginBottom": "5px",
            "textAlign": "center",
            "width": "13.6%",
          }
        }
      >
        <button
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "transparent",
              "border": "none",
              "color": "#ffffff",
            }
          }
        >
          2
        </button>
      </li>
      <li
        style={
          Object {
            "display": "inline-block",
            "listStyleType": "none",
            "marginBottom": "5px",
            "textAlign": "center",
            "width": "13.6%",
          }
        }
      >
        <button
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "transparent",
              "border": "none",
              "color": "#ffffff",
            }
          }
        >
          3
        </button>
      </li>
      <li
        style={
          Object {
            "display": "inline-block",
            "listStyleType": "none",
            "marginBottom": "5px",
            "textAlign": "center",
            "width": "13.6%",
          }
        }
      >
        <button
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "transparent",
              "border": "none",
              "color": "#ffffff",
            }
          }
        >
          4
        </button>
      </li>
      <li
        style={
          Object {
            "display": "inline-block",
            "listStyleType": "none",
            "marginBottom": "5px",
            "textAlign": "center",
            "width": "13.6%",
          }
        }
      >
        <button
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "transparent",
              "border": "none",
              "color": "#ffffff",
            }
          }
        >
          5
        </button>
      </li>
      <li
        style={
          Object {
            "display": "inline-block",
            "listStyleType": "none",
            "marginBottom": "5px",
            "textAlign": "center",
            "width": "13.6%",
          }
        }
      >
        <button
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "transparent",
              "border": "none",
              "color": "#ffffff",
            }
          }
        >
          6
        </button>
      </li>
      <li
        style={
          Object {
            "display": "inline-block",
            "listStyleType": "none",
            "marginBottom": "5px",
            "textAlign": "center",
            "width": "13.6%",
          }
        }
      >
        <button
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "transparent",
              "border": "none",
              "color": "#ffffff",
            }
          }
        >
          7
        </button>
      </li>
      <li
        style={
          Object {
            "display": "inline-block",
            "listStyleType": "none",
            "marginBottom": "5px",
            "textAlign": "center",
            "width": "13.6%",
          }
        }
      >
        <button
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "transparent",
              "border": "none",
              "color": "#ffffff",
            }
          }
        >
          8
        </button>
      </li>
      <li
        style={
          Object {
            "display": "inline-block",
            "listStyleType": "none",
            "marginBottom": "5px",
            "textAlign": "center",
            "width": "13.6%",
          }
        }
      >
        <button
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "transparent",
              "border": "none",
              "color": "#ffffff",
            }
          }
        >
          9
        </button>
      </li>
      <li
        style={
          Object {
            "display": "inline-block",
            "listStyleType": "none",
            "marginBottom": "5px",
            "textAlign": "center",
            "width": "13.6%",
          }
        }
      >
        <button
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "transparent",
              "border": "none",
              "color": "#ffffff",
            }
          }
        >
          10
        </button>
      </li>
      <li
        style={
          Object {
            "display": "inline-block",
            "listStyleType": "none",
            "marginBottom": "5px",
            "textAlign": "center",
            "width": "13.6%",
          }
        }
      >
        <button
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "transparent",
              "border": "none",
              "color": "#ffffff",
            }
          }
        >
          11
        </button>
      </li>
      <li
        style={
          Object {
            "display": "inline-block",
            "listStyleType": "none",
            "marginBottom": "5px",
            "textAlign": "center",
            "width": "13.6%",
          }
        }
      >
        <button
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "transparent",
              "border": "none",
              "color": "#ffffff",
            }
          }
        >
          12
        </button>
      </li>
      <li
        style={
          Object {
            "display": "inline-block",
            "listStyleType": "none",
            "marginBottom": "5px",
            "textAlign": "center",
            "width": "13.6%",
          }
        }
      >
        <button
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "transparent",
              "border": "none",
              "color": "#ffffff",
            }
          }
        >
          13
        </button>
      </li>
      <li
        style={
          Object {
            "display": "inline-block",
            "listStyleType": "none",
            "marginBottom": "5px",
            "textAlign": "center",
            "width": "13.6%",
          }
        }
      >
        <button
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "transparent",
              "border": "none",
              "color": "#ffffff",
            }
          }
        >
          14
        </button>
      </li>
      <li
        style={
          Object {
            "display": "inline-block",
            "listStyleType": "none",
            "marginBottom": "5px",
            "textAlign": "center",
            "width": "13.6%",
          }
        }
      >
        <button
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "transparent",
              "border": "none",
              "color": "#ffffff",
            }
          }
        >
          15
        </button>
      </li>
      <li
        style={
          Object {
            "display": "inline-block",
            "listStyleType": "none",
            "marginBottom": "5px",
            "textAlign": "center",
            "width": "13.6%",
          }
        }
      >
        <button
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "transparent",
              "border": "none",
              "color": "#ffffff",
            }
          }
        >
          16
        </button>
      </li>
      <li
        style={
          Object {
            "display": "inline-block",
            "listStyleType": "none",
            "marginBottom": "5px",
            "textAlign": "center",
            "width": "13.6%",
          }
        }
      >
        <button
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "transparent",
              "border": "none",
              "color": "#ffffff",
            }
          }
        >
          17
        </button>
      </li>
      <li
        style={
          Object {
            "display": "inline-block",
            "listStyleType": "none",
            "marginBottom": "5px",
            "textAlign": "center",
            "width": "13.6%",
          }
        }
      >
        <button
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "transparent",
              "border": "none",
              "color": "#ffffff",
            }
          }
        >
          18
        </button>
      </li>
      <li
        style={
          Object {
            "display": "inline-block",
            "listStyleType": "none",
            "marginBottom": "5px",
            "textAlign": "center",
            "width": "13.6%",
          }
        }
      >
        <button
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "transparent",
              "border": "none",
              "color": "#ffffff",
            }
          }
        >
          19
        </button>
      </li>
      <li
        style={
          Object {
            "display": "inline-block",
            "listStyleType": "none",
            "marginBottom": "5px",
            "textAlign": "center",
            "width": "13.6%",
          }
        }
      >
        <button
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "transparent",
              "border": "none",
              "color": "#ffffff",
            }
          }
        >
          20
        </button>
      </li>
      <li
        style={
          Object {
            "display": "inline-block",
            "listStyleType": "none",
            "marginBottom": "5px",
            "textAlign": "center",
            "width": "13.6%",
          }
        }
      >
        <button
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "transparent",
              "border": "none",
              "color": "#ffffff",
            }
          }
        >
          21
        </button>
      </li>
      <li
        style={
          Object {
            "display": "inline-block",
            "listStyleType": "none",
            "marginBottom": "5px",
            "textAlign": "center",
            "width": "13.6%",
          }
        }
      >
        <button
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "transparent",
              "border": "none",
              "color": "#ffffff",
            }
          }
        >
          22
        </button>
      </li>
      <li
        style={
          Object {
            "display": "inline-block",
            "listStyleType": "none",
            "marginBottom": "5px",
            "textAlign": "center",
            "width": "13.6%",
          }
        }
      >
        <button
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "transparent",
              "border": "none",
              "color": "#ffffff",
            }
          }
        >
          23
        </button>
      </li>
      <li
        style={
          Object {
            "display": "inline-block",
            "listStyleType": "none",
            "marginBottom": "5px",
            "textAlign": "center",
            "width": "13.6%",
          }
        }
      >
        <button
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "transparent",
              "border": "none",
              "color": "#ffffff",
            }
          }
        >
          24
        </button>
      </li>
      <li
        style={
          Object {
            "display": "inline-block",
            "listStyleType": "none",
            "marginBottom": "5px",
            "textAlign": "center",
            "width": "13.6%",
          }
        }
      >
        <button
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "transparent",
              "border": "none",
              "color": "#ffffff",
            }
          }
        >
          25
        </button>
      </li>
      <li
        style={
          Object {
            "display": "inline-block",
            "listStyleType": "none",
            "marginBottom": "5px",
            "textAlign": "center",
            "width": "13.6%",
          }
        }
      >
        <button
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "transparent",
              "border": "none",
              "color": "#ffffff",
            }
          }
        >
          26
        </button>
      </li>
      <li
        style={
          Object {
            "display": "inline-block",
            "listStyleType": "none",
            "marginBottom": "5px",
            "textAlign": "center",
            "width": "13.6%",
          }
        }
      >
        <button
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "transparent",
              "border": "none",
              "color": "#ffffff",
            }
          }
        >
          27
        </button>
      </li>
      <li
        style={
          Object {
            "display": "inline-block",
            "listStyleType": "none",
            "marginBottom": "5px",
            "textAlign": "center",
            "width": "13.6%",
          }
        }
      >
        <button
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "transparent",
              "border": "none",
              "color": "#ffffff",
            }
          }
        >
          28
        </button>
      </li>
      <li
        style={
          Object {
            "display": "inline-block",
            "listStyleType": "none",
            "marginBottom": "5px",
            "textAlign": "center",
            "width": "13.6%",
          }
        }
      >
        <button
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "transparent",
              "border": "none",
              "color": "#ffffff",
            }
          }
        >
          29
        </button>
      </li>
      <li
        style={
          Object {
            "display": "inline-block",
            "listStyleType": "none",
            "marginBottom": "5px",
            "textAlign": "center",
            "width": "13.6%",
          }
        }
      >
        <button
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "transparent",
              "border": "none",
              "color": "#ffffff",
            }
          }
        >
          30
        </button>
      </li>
      <li
        style={
          Object {
            "display": "inline-block",
            "listStyleType": "none",
            "marginBottom": "5px",
            "textAlign": "center",
            "width": "13.6%",
          }
        }
      >
        <button
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "transparent",
              "border": "none",
              "color": "#ffffff",
            }
          }
        >
          31
        </button>
      </li>
    </ul>
  </div>
</div>
`;
